### DISCUSSION TITLE
Consistent Object Initializer Order

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/csharplang#8096] [Language Ideas]
Is there a way for me to set a priority/order on properties so that I can refactor my app and have all object initializer properties set in the same consistent order?

**Comment by HaloFour** [comment]
Having the object initializers execute in anything but lexical order would be a nightmare to reason about.  If you have a preferred order I would suggest an analyzer with a fixer that would rewrite the initializers in the expected order.  Personally I think that if there is a required order in which to invoke something the API should force that through something like a series of builder interfaces.

**Reply by TonyValenti** [reply]
Do you know of any existing analyzers/fixers I could use to do this?

**Reply by CyrusNajmabadi** [reply]
it would be fairly trivial to write one.  I don't believe anything exists as this pattern is definitely discouraged.  Having suchan ordering dependency is not desirable  from the language's perspective.  If such a dependency is something wanted, using a pattern like a 'builder' pattern would be encouraged so that the user can only initialize the values they're in the right initialization state for.

**Reply by TonyValenti** [reply]
Hi @CyrusNajmabadi -
This is merely for code aesthetics. For example, if I had an Address class, would want properties in this order:

Name
Street1
Street2
City
Region
PostalCode 
Country

And if someone had an order like this:
PostalCode
Region
Street2
City
Name
Country
Street1

I'd like to be able to quickly refactor to my desired order.

**Reply by CyrusNajmabadi** [reply]
Got it.  would be easy to write an analyzer/fixer for this then :)

**Reply by TonyValenti** [reply]
Easy if you know how to do it!
Is there anyone here who would be interested in getting paid to do this for me?  If so, post here and I'll set up a time to chat.
